### PR TITLE
Add uuid field to OgMembership

### PR DESCRIPTION
**Pull request by MPParsley**
Solves #425 & #536.
- Added uuid field to OgMembership;
- Added hook_update to update existing installations;

### CODE REVIEW

File: og.install [hunk 1]
```diff
@@ -77,3 +77,19 @@ function og_update_8001(&$sandbox) {
...
+ */
+function og_update_8002() {
+  $manager = \Drupal::entityDefinitionUpdateManager();
+  $entity_type = $manager->getEntityType('og_membership');
+  $entity_keys = $entity_type->getKeys();
+  $entity_keys['uuid'] = 'uuid';
+  $entity_type->set('entity_keys', $entity_keys);
+  $manager->updateEntityType($entity_type);
+
+  foreach (['uuid'] as $field_name) {
```
Comment (zerolab) [inline og.install]: this should really only be `$manager->updateFieldStorageDefinition($manager->getFieldStorageDefinition('uuid', 'og_membership'));`

File: og.install [hunk 2]
```diff
@@ -77,3 +77,17 @@ function og_update_8001(&$sandbox) {
...
+ * Add uuid field to OgMembership.
+ */
+function og_update_8002() {
+  $manager = \Drupal::entityDefinitionUpdateManager();
+  $entity_type = $manager->getEntityType('og_membership');
+  $entity_keys = $entity_type->getKeys();
+  $entity_keys['uuid'] = 'uuid';
+  $entity_type->set('entity_keys', $entity_keys);
+  $manager->updateEntityType($entity_type);
+
```
Comment (MPParsley) [inline og.install]:
```suggestion
```

File: og.install (hunk 1 above)
Comment (MPParsley) [reply og.install]: Indeed.

**Comment by MPParsley** [comment]
Tests are broken due to #533.

**Comment by pfrenssen** [comment]
This looks good to me. I tested the update hook against my production database with 60k memberships and it works fine:

```
> select id, uuid from og_membership limit 10;
+-------+--------------------------------------+
| id    | uuid                                 |
+-------+--------------------------------------+
| 20818 | 00026af5-6f97-48f4-abd8-767c8ac88ee7 |
| 36219 | 00027df5-adb8-4cc9-a7be-e96a14780ad8 |
|  7555 | 0005eb39-505a-4afa-b7d2-a5b0b945bc5e |
| 18904 | 0006b6ba-1514-416f-b5c8-b38a29ff37e8 |
| 46627 | 0007ade9-36b9-43d6-a1bb-9c14a56eae2b |
|  8361 | 0007b802-66c1-46c8-8fbf-d4b544abd343 |
| 52125 | 0007bfd5-84c2-4e6b-9af1-b4f7e6378e2d |
| 64321 | 00084eb8-4222-4a4c-ba24-59b6b0160053 |
| 17820 | 000995b1-5e54-44af-b68b-5fa4acb8187c |
| 55504 | 000a35ff-677d-4ecf-a35c-aa90aed19ee4 |
+-------+--------------------------------------+
10 rows in set (0.001 sec)
```

I'm not sure though why we would need to have a UUID associated with a membership, since we need to do a lot more work in order to be able to sync memberships between environments, like validating the existence of the users, the groups, the roles, and updating the membership ID to match the new environment. Is this intended to be the first step to supporting migrations between environments?

Some more information on the motivation for this change would be welcome.

**Review by pfrenssen — APPROVED** [review]
Let's get this in. This is a required step in order to make syncing memberships between environments possible in the future, and it helps to move #231 along.

**Comment by MPParsley** [comment]
> Some more information on the motivation for this change would be welcome.

Sorry, I don't know the initial reason this was added but indeed, it may become useful at a given time in the future.

**Pull request merged**